SunEngine.Core/Presenters/SearchPresenter.cs

```diff
@@ -30,6 +34,40 @@ public Task<UserInfoView[]>  SearchByUsernameOrLink(string searchString)
                 Name = x.UserName
             }).ToArrayAsync();
         }
+
+        public Task<MaterialSearchInfoView[]> SearchByMaterials(string searchString)
+        {
+            switch (db.NameSqlProvider)
```

Comment:
You need to move the query predicate logic for all databases to a separate service (or at least a private method).